### DISCUSSION TITLE
Fast execution helpers?

### DISCUSSION THREAD

**Discussion by jabruder** [psycopg/psycopg#349] [Q&A] [answered]
Hello all,
Is there a version of the [fast execution helpers](https://www.psycopg.org/docs/extras.html#fast-execution-helpers) from Psycopg 2 in Psycopg 3? Or, should I be looking at asyncio to speed up updating/inserting about a 1,000 records at a time?
Thank you!

**Comment by dvarrazzo** [comment, marked as answer]
@jabruder in a few days we will release psycopg 3.1 which includes the pipeline mode, which will add a huge boost to execution batches.

What the "fast execution helpers" do is just to [concatenate several copies of the query](https://github.com/psycopg/psycopg2/blob/3e7bb8d1aa631320234f9e05ef940a1d1e9a627f/lib/extras.py#L1165) and execute that. It should be easy to reproduce the same in your code.
